lib/matplotlib/collections.py

```diff
@@ -1127,14 +1122,40 @@ def legend_elements(self, prop="colors", num="auto",
             raise ValueError("Valid values for `prop` are 'colors' or "
                              f"'sizes'. You supplied '{prop}' instead.")
 
-        fmt.set_bounds(func(u).min(), func(u).max())
+        func_value = np.asarray(func(u))
+        func_is_numeric = np.issubdtype(func_value.dtype, np.number)
+
+        if fmt is None:
+            if func_is_numeric:
+                fmt = mpl.ticker.ScalarFormatter(
+                    useOffset=False, useMathText=True
+                )
+            else:
+                fmt = mpl.ticker.StrMethodFormatter("{x}")
+        elif isinstance(fmt, str):
+            fmt = mpl.ticker.StrMethodFormatter(fmt)
+        fmt.create_dummy_axis()
+
+        if func_is_numeric:
+            fmt.set_bounds(min(func_value), max(func_value))
+
         if num == "auto":
             num = 9
             if len(u) <= num:
                 num = None
         if num is None:
             values = u
             label_values = func(values)
+        elif not func_is_numeric:
+            # Values are not numerical so instead of interpolating
+            # just choose evenly distributed indexes instead:
+            def which_idxs(m, n):
+                out = np.rint(np.linspace(1, n, min(m, n)) - 1)
+                return out.astype(int)
+            label_values = func(u)
+            cond = which_idxs(num, len(label_values))
+            values = u[cond]
+            label_values = label_values[cond]
```

Comment:
`num` is limited to 9 by default and this code path was simply added for consistency with that. But the user can choose to opt out of it by setting `num=None` and all entries will be displayed, just like when using numerics.

I'm a little sceptic on subsampling too, but for numerics as well. Because it doesn't always make sense to interpolate. Examples that comes to mind are bool-likes, integers and the id-number trick we're using in our examples. But matplotlib doesn't have any strong opinions on that though and instead lets the user choose.